pyproject.toml

```diff
@@ -57,6 +57,9 @@ dependencies = [
     "numba>=0.59; python_version > '3.11'",
     "numba>=0.57; python_version <= '3.11'",
     "pymatching>=2.2.1",
+    "qsample>=0.0.2",
```

Comment:
Just a quick comment here, because I stumbled over it.
This package hardly seems maintained and is not really well packaged (as is kind of obvious from all the special handling you have to do here).
Do you really need that dependency for the feature you are trying to add here? Or would there be another way to eliminate that dependency and maybe vendor the respective code as part of QECC?

Reply:
Good point, I already had discussions with Tom on it. Problem is: Without the package most of the code/features will not work (in particular the whole simulation part).
If you prefer one could consider moving the simulation part to an "untested/unimplemented" directory or Zenodo and only keep the construction part which does not rely on qSample.